client/src/js/components/courseEvals/CourseEvals.js

```diff
@@ -0,0 +1,300 @@
+  // let relatedCourses = coursesList.filter(c => c.course_code === courseCode && getCourseKey(c) !== thisCourseKey);
+
+  return [
+    {
+      "_id":"2603033019012020260303301926",
+      "course_code": 'INFP31ZL',
+      "year":2020,
+      "term":"fallSem",
+      "title":"Software Engineering B",
+      "title_jp":"ソフトウェア工学B",
+      "instructor":"HAGIMOTO, Junzo/HIRANABE, Kenji/FUKAZAWA, Yoshiaki/WASHIZAKI, Hironori",
+      "instructor_jp":"萩本 順三/平鍋 健児/深澤 良彰/鷲崎 弘宜",
+      "occurrences":[
+        {
+           "day":3,
+           "start_period":5,
+           "end_period":5,
+           "building":"-1",
+           "classroom":"undecided",
+           "location":"-1-undecided"
+        }
+      ],
+      "lang":"JP",
+      "keys":[
+        {
+           "school":"FSE",
+           "key":"2603033019012020260303301926"
+        }
+      ]
+    },
+    {
+      "_id":"26GC02300301202026MC03300926",
+      "course_code": 'INFP31ZL',
+      "year":2020,
+      "term":"fallSem",
+      "title":"Software Engineering",
+      "title_jp":"Software Engineering",
+      "instructor":"ARCAINI, Paolo/KOBAYASHI, Tsutomu/FUKAZAWA, Yoshiaki/HONIDEN, Shinichi/WASHIZAKI, Hironori",
+      "instructor_jp":"アルカイニ パオロ/小林 努/深澤 良彰/本位田 真一/鷲崎 弘宜",
+      "occurrences":[
+        {
+           "day":1,
+           "start_period":3,
+           "end_period":3,
+           "building":"-1",
+           "classroom":"undecided",
+           "location":"-1-undecided"
+        }
+      ],
+      "keywords":[
+        "IPSE"
+      ],
+      "lang":"EN",
+      "keys":[
+        {
+           "school":"FSE",
+           "key":"26GC02300301202026MC03300926"
+        }
+      ]
+    }
+  ];
+}
+
+const getCourseEvals = (evaluations, courseKey) => evaluations.filter(e => e.course_key === courseKey);
+
+class CourseEvals extends React.Component {
+  state = {
+    thisCourse: {},
+    relatedCourses: [],
+    evaluations: [],
+    thisCourseEvals: [],
+    satisfaction: 0,
+    difficulty: 0,
+    benefit: 0,
+    isLoaded: false
+  };
+
+  async componentDidMount () {
+    const courseID = qs.parse(this.props.location.search, { ignoreQueryPrefix: true }).courseID;
+    const coursesList = await getAllCourses();
+    const thisCourse = getCourse(coursesList, courseID);
+    const thisCourseCode = getCourseCode(thisCourse);
+    const thisCourseKey = getCourseKey(thisCourse);
+    const relatedCourses = getRelatedCourses(coursesList, thisCourseCode, thisCourseKey);
+    const evaluations = await getAllEvals(thisCourseCode);
+    const thisCourseEvals = getCourseEvals(evaluations, thisCourseKey);
+    let satisfactionSum = 0, difficultySum = 0, benefitSum = 0;
+    thisCourseEvals.forEach(evaluation => {
+      satisfactionSum += evaluation.satisfaction;
+      difficultySum += evaluation.difficulty;
+      benefitSum += evaluation.benefit;
+    });
+    const satisfaction = Math.round(satisfactionSum / thisCourseEvals.length * 2) / 2;
```

Comment:
Interesting way to round to the nearest .5!
It'd be helpful if you can leave a code comment on top of it